docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,118 @@
+********************************
+Developer Getting Started Guide
+********************************
+
+This page describes how to setup your development environment to:
+
+* Install VS Code and Inmanta extension.
+* Setting up Python virtual environments.
+* Benefit from linting and code navigation by setting up a project.
+* Set project sources
+* Module developers guide
+
+
+Install VS Code and Inmanta extension
+#######################################
+
+This step does not need any additional information but if you need some help, you can always ask a colleague.
+
+In order to install VS Code, you can refer to `this <https://code.visualstudio.com/learn/get-started/basics>`_ page.
+
+Further information about Inmanta VS Code extension is available on `this <https://github.com/inmanta/vscode-inmanta>`_ page.
+
+
+Setting up Python virtual environments
+########################################
+
+Python virtual environments have to be created using ``venv`` by running:
+
+.. code-block:: bash
+    
+    python3 -m venv ~/.virtualenvs/<env_name>
+
+Then activated by running:
+
+.. code-block:: bash
+    
+    source ~/.virtualenvs/<env_name>/bin/activate
+
+**Upgrading your ``pip`` will save you a lot of time and troubleshooting (due to changes in the pip resolver in version 20 and 21).** you can do so by running:
+
+.. code-block:: bash
+    
+    pip install --upgrade pip
+
+Usually the projects that you will work on, come with ``requirements.txt`` or ``requirements.dev.txt`` to install the required modules, if not, it is a good idea to ``pip install``:
```

Comment:
About the packages, two remarks:

1. I would split this into two very clear parts: 
   1. if you work from an existing project then these commands
   2. otherwise, these commands 
2. because of that, I would move to after the project install part of the guide.